samples/AspNetIntegration/FooMiddleware.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Middleware;
+
+namespace AspNetIntegration
+{
+    internal class FooMiddleware : IFunctionsWorkerMiddleware
+    {
+        //<docsnippet_aspnet_middleware>
+        public Task Invoke(FunctionContext context, FunctionExecutionDelegate next)
+        {
+            // retrieve http context from function context
+            HttpContext httpContext = context.GetHttpContext() 
+                ?? throw new InvalidOperationException($"{nameof(context)} has no http context associated with it.");
```

Comment:
Question: Should we throw here or gracefully exit? Because there is nothing stopping from customers to add a non http trigger today this function app and it will throw here. 

May be leave a comment here about that or register the middleware only for http requests like [here](https://github.com/Azure/azure-functions-dotnet-worker/blob/a6322a3eb70bb6df93a9b22950c7bd50f8157e97/samples/CustomMiddleware/Program.cs#L23)

Reply:
good callout - will register this only for http requests.